internal/k8s/controllers/layer2_status_controller.go

```diff
@@ -61,32 +67,54 @@ func (r *Layer2StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	level.Info(r.Logger).Log("controller", "Layer2StatusReconciler", "start reconcile", req.NamespacedName.String())
 	defer level.Info(r.Logger).Log("controller", "Layer2StatusReconciler", "end reconcile", req.NamespacedName.String())
 
-	svcName := strings.TrimSuffix(req.Name, fmt.Sprintf("-%s", r.NodeName))
-	ipAdvS := r.StatusFetcher(types.NamespacedName{Name: svcName, Namespace: req.Namespace})
+	serviceName, serviceNamespace := req.Name, req.Namespace
+	statusName, statusNamespace := utilrand.String(6), r.Namespace
+
+	ipAdvS := r.StatusFetcher(types.NamespacedName{Name: serviceName, Namespace: serviceNamespace})
+	stateList := &v1beta1.ServiceL2StatusList{}
+	if err := r.Client.List(ctx, stateList, client.MatchingFields{
+		serviceIndexName: types.NamespacedName{Name: serviceName, Namespace: serviceNamespace}.String(),
+	}); err != nil {
+		return ctrl.Result{}, err
+	}
 
+	var err error
 	if len(ipAdvS) == 0 {
-		err := r.Client.Delete(ctx, &v1beta1.ServiceL2Status{ObjectMeta: metav1.ObjectMeta{Name: req.Name, Namespace: req.Namespace}})
+		for _, state := range stateList.Items {
+			err = r.Client.Delete(ctx, &state)
+		}
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	state := &v1beta1.ServiceL2Status{ObjectMeta: metav1.ObjectMeta{
-		Name:      req.Name,
-		Namespace: req.Namespace,
-	}}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: req.Name, Namespace: req.Namespace}, state)
-	if err != nil && !errors.IsNotFound(err) {
-		return ctrl.Result{}, err
+	var state *v1beta1.ServiceL2Status
+	if len(stateList.Items) == 0 {
+		state = &v1beta1.ServiceL2Status{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      statusName,
```

Comment:
We don't really need to generate the string randomly ourselves, we can just use here `GenerateName: "l2-"` instead of `Name` :)

Reply:
Oh, forgot that